py/desitarget/lyazcat.py

```diff
@@ -138,7 +140,10 @@ def make_new_zcat(zbestname, qn_flag=False, sq_flag=False, abs_flag=False,
     # ADM also add the appropriate bit-columns.
     Mxcols, _, _, = main_cmx_or_sv(fms, scnd=True)
     for col in Mxcols:
-        zcat[col] = fms[zid][col]
+        if col in fms.dtype.names:
+            zcat[col] = fms[zid][col]
+        else:
+            log.error(f'Input fibermap missing {col}; leaving it blank')
```

Comment:
Consider, here, failing spectacularly with 
```
log.error(f'Input fibermap missing {col}; leaving it blank')
if col in zcatdatamodel.dtype.names:
    raise ValueError('{col} is required in the data model!')
```

Reply:
Actually, the only columns that are strictly needed for the MTL loop are:
```
from desitarget.mtl import zcatdatamodel as mtlzdm, survey_data_model
required_cols = survey_data_model(mtlzdm, survey="main").dtype.names
```
The other columns (such as the `DESI_/BGS_/MWS_/SCND_TARGET` columns and `SPECTYPE`) were added for "general utility" after a lengthy discussion about the `zqsocat` format on a  DESI data telecon. So, you might want to just check against `required_cols` for a spectacular fail.